projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -320,6 +327,33 @@ def pre_ingest_process(self, payload: List[dict]) -> List[dict]:
         :param metadata: Optional[IngestionMetadata] dict object containing
         metadata produced by the pre-ingest plugin, and possibly used by
         other pre-ingest, ingest, or post-ingest plugins.
+        :param destination_table: Optional[string]
+            (Optional) The name of the table, where the data should be
+            ingested into. This argument is optional, and needs to be
+            considered only when the name of the destination table is not
+            included in the payload itself.
+            NOTE: A read-only parameter. It is not expected to be modified and
+            returned.
+        :param target: Optional[string]
+            (Optional) Used to identify where the data should be ingested into.
+                Specifies a data source and its destination database.
+                The values for this parameter can be in the format
+                `<some-data-source_and-db-table>`, or as a URL.
+                Example:
+                      http://example.com/<some-api>/<data-source_and-db-table>
+            This parameter does not need to be used, in case the
+            `INGEST_TARGET_DEFAULT` environment variable is set. This can be
+            made by plugins, which may set default value, or it can be
+            overwritten by users.
+            NOTE: A read-only parameter. It is not expected to be modified and
+            returned.
+        :param collection_id: string
+            (Optional) An identifier to indicate that data from different
+            method invocations belong to same collection. Defaults to
+            "data_job_name|OpID", meaning all method invocations from a data
+            job run will belong to the same collection.
+            NOTE: A read-only parameter. It is not expected to be modified and
+            returned.
```

Comment:
maybe include documentation on expectation if an eventual exception is raised by a pre-processor, regarding ingestion itself

Reply:
Is it necessary? If something happens during pre-processing, ingestion should be blocked to avoid data corruption.

Reply:
It seems fair, if any pre-processor in the sequence configured fails, the ingestion workflow should be interrupted;
This I think should be explicitly added to pre-processor documentation - what happens if an exception is raised - and maybe a note to ingestion docs also

Reply:
Documentation updated. Thank you